### PR TITLE
Updates for CI

### PR DESCRIPTION
**Pull request by laurenchilutti**
**Description**

Changes to the CI files.  Specifically mirroring improvements made to the GFDL_atmos_cubed_sphere CI.  Moving this CI to a new Parallelworks cloud cluster.  Reorganizing the file structure on Parallelworks.  Adding conditions so that the CI does not run in personal forks.

Fixes # (issue)

**How Has This Been Tested?**

This will be tested by this PR.

**Checklist:**

Please check all whether they apply or not
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/Intel_Parallelworks_CI.yaml [hunk 1]
```diff
@@ -13,21 +13,29 @@ on:
   pull_request:
     branches:
       - main
+
+#this should cancel in progress ci runs for the same PR 
+#(e.g. a second commit on the same PR comes in while CI is still running)
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```
Comment (bensonr) [inline .github/workflows/Intel_Parallelworks_CI.yaml:21]: Do all of the actions below wait until confirmation of a cancellation?  I guess I'm concerned a cancellation is ignored and the updated actions continue concurrently in the separate SHA macro-based workspace.

File: .github/workflows/Intel_Parallelworks_CI.yaml [hunk 2]
```diff
@@ -94,15 +103,15 @@ jobs:
           - d96_500m.solo.mtn_schar.sh
     steps:
       # This will end the slurm job started in the checkout job
-      - run: scancel -n SP_$GITHUB_REF
+      - run: scancel -n $GITHUB_SHA
```
Comment (bensonr) [inline .github/workflows/Intel_Parallelworks_CI.yaml:106]: Why is this needed?  Shouldn't the checkout slurm job terminate on it's own correctly?

File: .github/workflows/Intel_Parallelworks_CI.yaml (hunk 1 above)
Comment (laurenchilutti) [reply .github/workflows/Intel_Parallelworks_CI.yaml:21]: I could give this a test.  In theory, it is supposed to only allow one run in the concurrency group to go at a time and the most recent run will trigger a cancel of the older run.

File: .github/workflows/Intel_Parallelworks_CI.yaml (hunk 2 above)
Comment (laurenchilutti) [reply .github/workflows/Intel_Parallelworks_CI.yaml:106]: I have this specific slurm job running a sleep command for 20 minutes.  If the checkout and build steps finish before 20 minutes, I would like to cancel that job.  It then frees up the nodes to then be assigned to the tests that run next.

**Review by bensonr — APPROVED** [review]
